src/wp-includes/class-wp-metadata-lazyloader.php

```diff
@@ -133,15 +133,19 @@ public function reset_queue( $object_type ) {
 	 * @since 4.5.0
 	 *
 	 * @param mixed $check The `$check` param passed from the 'get_term_metadata' hook.
+	 * @param int   $object_id ID of the object metadata is for.
 	 * @return mixed In order not to short-circuit `get_metadata()`. Generally, this is `null`, but it could be
 	 *               another value if filtered by a plugin.
 	 */
-	public function lazyload_term_meta( $check ) {
+	public function lazyload_term_meta( $check, $object_id ) {
 		if ( ! empty( $this->pending_objects['term'] ) ) {
-			update_termmeta_cache( array_keys( $this->pending_objects['term'] ) );
+			$object_ids = array_keys( $this->pending_objects['term'] );
+			if ( in_array( $object_id, $object_ids, true ) ) {
+				update_termmeta_cache( $object_ids );
 
-			// No need to run again for this set of terms.
-			$this->reset_queue( 'term' );
+				// No need to run again for this set of terms.
+				$this->reset_queue( 'term' );
+			}
```

Comment:
See above, we need this logic to still work the same as before _if_ no `$object_id` is provided, for BC.